express-app/app.models.js

```diff
@@ -21,33 +21,52 @@ exports.fetchArticleID = (article_id) => {
   });
 };
 
-exports.fetchArticles = (sort_by = 'created_at', order = 'DESC') => {
+exports.fetchArticles = (sort_by, order, topic) => {
   const validSortColumns = ['author', 'title', 'created_at', 'votes'];
   const validOrder = ['ASC', 'DESC'];
+  const validTopics = ['mitch', 'cats', 'paper'];
 
-  if (!validSortColumns.includes(sort_by)) {
-    return Promise.reject({
-      status: 400,
-      msg: 'Bad Request: Invalid Sort Query',
-    });
+  let query = `
+  SELECT articles.*, COUNT(comments.comment_id) AS comment_count
+  FROM articles
+  LEFT JOIN comments ON comments.article_id = articles.article_id
+  `;
+
+  if (topic) {
```

Comment:
You shouldn't need this outer if statement - if topic doesn't exist then the condition below will always be false